packages/common/src/codegen/render-solidity/renderTypeHelpers.ts

```diff
@@ -10,7 +10,7 @@ export function renderTypeHelpers(options: { fields: RenderField[]; primaryKeys:
   }
 
   // bool is special - it's the only primitive value type that can't be typecasted to/from
-  if (fields.some(({ typeId }) => typeId === "bool")) {
+  if (fields.some(({ internalTypeId }) => internalTypeId.match("bool"))) {
```

Comment:
for context - why this change?

Reply:
`.match` - because arrays may need their items converted, and this is simpler than `["bool", "bool[]"].includes` and might include something else I forgot
`internal` - if I add more userTypes a bool could be wrapped, so typeId wouldn't help here